pkg/services/ngalert/models/alert_rule.go

```diff
@@ -367,13 +368,21 @@ func (alertRule *AlertRule) GetLabels(opts ...LabelOption) map[string]string {
 }
 
 func (alertRule *AlertRule) GetEvalCondition() Condition {
+	meta := map[string]string{
+		"Name":    alertRule.Title,
+		"Uid":     alertRule.UID,
+		"Type":    string(alertRule.Type()),
+		"Version": strconv.FormatInt(alertRule.Version, 10),
```

Comment:
WDYT about just having this be a straight up map of headers, and using the full header name as the key?

The expression pipeline does the same in `expr.Request` even though the thing handling the request might not be contacted over HTTP and might not be out of process at all (TestData datasource) - it uses headers anyway. There's some precedent for it I think

Reply:
I do not have a strong opinion and can change it if we think that would be clearer\simpler.

 My reasons for having conversion in the intermediate layer between the scheduler and expressions service are the following: 
- Calling it `headers` seems too specific because it exposes implementation details of the service domain (expression engine), which the eval package is supposed to isolate.
- Calling it headers also would imply that I would have to use header-specific syntax like `X-Rule-Uid` for keys, which apart from looking awkward, limits usage of this data only in the headers. However, I think in the case when the query service is a micro-service, this information could be provided downstream in a format other than headers where it could be used for enriching logging context or tracing information. 
- This would also assume that the scheduler (or other places that specify the "header") should properly prepare the map to be compatible with HTTP specifications but if the expression service changes the API, there may be other mechanisms to provide request metadata.

Considering all the above + the fact that this change introduces a new term in the domain of rule evaluators, I decided to make the field name neutral and leave it up to eval package implementation to decide what to do with the information.